src/wechaty_puppet_hostie/puppet.py

```diff
@@ -195,22 +195,23 @@ async def room_list(self) -> List[str]:
             raise WechatyPuppetGrpcError('can"t get room_list response')
         return response.ids
 
-    async def message_image(self, message_id: str, image_type: ImageType
+    async def message_image(self, message_id: str, image_type: ImageType = 3
                             ) -> FileBox:
         """
         get message image data
         :param message_id:
         :param image_type:
         :return:
         """
-        response = await self.puppet_stub.message_image(
-            id=message_id,
-            type=image_type)
-        if response is None:
-            # TODO -> need to refactor the raised error
-            raise WechatyPuppetGrpcError('response is invalid')
+        response = await self.puppet_stub.message_image(id=message_id, type=image_type)
         json_response = json.loads(response.filebox)
-        return FileBox.from_json(obj=json_response)
+        if 'base64' not in json_response:
+            raise WechatyPuppetGrpcError('image response data structure is not correct')
+        file_box = FileBox.from_base64(
+            json_response['base64'],
+            name=json_response['name'] + '.png'
```

Comment:
here's an attention, the picture file extension is renamed to '.png', the api response['name'] is 'xxxxxxx.bat', if the naming is not images extension, it will send a file so that I renamed it to '.png'.

Reply:
latest grpc fix the api. The parameter 'name' is correct. BTW, if message is a 'gif', WeChat will send it as an 'EMOTICON'.
below is detail response from api:
```
<msg>
	<emoji fromusername="loveheaven520" 
	       tousername="17426117522@chatroom" 
		   type="2" 
		   idbuffer="media:0_0" md5="2a4ec5b96b3587ed04a28475de5760f5" 
		   len="2338425" 
		   productid="" 
		   androidmd5="2a4ec5b96b3587ed04a28475de5760f5" 
		   androidlen="2338425" 
		   s60v3md5="2a4ec5b96b3587ed04a28475de5760f5" 
		   s60v3len="2338425" 
		   s60v5md5="2a4ec5b96b3587ed04a28475de5760f5" 
		   s60v5len="2338425" 
		   cdnurl="http://emoji.qpic.cn/wx_emoji/sk7TwgqFeKuw5pghVlLlwzhGK4ZeUImYeySrJW9IzpBGicnlIiaHDAXSiaUh16n1tSf/" 
		   designerid="" 
		   thumburl="" 
		   encrypturl="http://emoji.qpic.cn/wx_emoji/sk7TwgqFeKuw5pghVlLlwzhGK4ZeUImYeySrJW9IzpBNlajpYibbPoa7RWxicXteS7/" 
		   aeskey="500a72127a60d9c00fb33d1447faafac" 
		   externurl="http://emoji.qpic.cn/wx_emoji/XsdZyT0o0NHuh9ZiaQHmLh8WjhLJ0xWBzdOa4ONuUf0GzZicHriafClEFib3vqKfOVr3/" externmd5="bc583d9413ae2e7828fb1041f7128146" 
		   width="335" 
		   height="404" 
		   tpurl="" 
		   tpauthkey="" 
		   attachedtext="" 
		   attachedtextcolor="" 
		   lensid=""
	>
	</emoji>
	<gameext type="0" 
	         content="0"
	>
	</gameext>
</msg>
```

we can use `cdnurl` to send the gif like:
`
file_box = FileBox.from_url(f'''{cdnurl}''', name='this_is_a_gif.gif')
            await conversation.say(file_box)
`